pbparam/optimisation_problems/data_fit.py

```diff
@@ -29,26 +29,19 @@ def update_simulation_parameters(simulation, parameter_values):
 
     return new_simulation
 
-
-def cost_function_full(simulation, map_inputs, scalings, data, variables_optimise, x):
+def cost_function_full(self, simulation, map_inputs, scalings, data, variables_optimise, x):
```

Comment:
If you pass `opt_problem` (see message in MS Teams), then all the other variables (except for `x`) should be called from there (e.g. `opt_problem.data`).